app/routing/DeparturesRouter.scala

```diff
@@ -20,22 +20,54 @@ import akka.stream.Materializer
 import akka.stream.scaladsl.Source
 import akka.util.ByteString
 import com.google.inject.Inject
+import controllers.V1DepartureMessagesController
 import controllers.V1DeparturesController
+import models.domain.DepartureId
+import models.domain.MessageId
 import play.api.mvc.Action
+import play.api.mvc.AnyContent
 import play.api.mvc.BaseController
 import play.api.mvc.ControllerComponents
+import play.api.mvc.PathBindable
 import v2.controllers.V2DeparturesController
 import v2.controllers.stream.StreamingParsers
+import v2.models.MovementId
+import v2.models.Bindings._
 
-class DeparturesRouter @Inject() (val controllerComponents: ControllerComponents, v1Departures: V1DeparturesController, v2Departures: V2DeparturesController)(
-  implicit val materializer: Materializer
+class DeparturesRouter @Inject() (
+  val controllerComponents: ControllerComponents,
+  v1Departures: V1DeparturesController,
+  v1DepartureMessages: V1DepartureMessagesController,
+  v2Departures: V2DeparturesController
+)(implicit
+  val materializer: Materializer
 ) extends BaseController
     with StreamingParsers
     with VersionedRouting {
 
   def submitDeclaration(): Action[Source[ByteString, _]] = route {
-    case Some("application/vnd.hmrc.2.0+json") => v2Departures.submitDeclaration()
-    case _                                     => v1Departures.submitDeclaration()
+    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE) => v2Departures.submitDeclaration()
+    case _                                                    => v1Departures.submitDeclaration()
+  }
+
+  def getMessage(departureId: String, messageId: String): Action[Source[ByteString, _]] = route {
+    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE) =>
+      (for {
+        convertedDepartureId <- implicitly[PathBindable[MovementId]].bind("departureId", departureId)
```

Comment:
I'm a bit confused about the use of movementId vs messageId vs departureId...Here we bind departureId to MovementId but we have the convertedMovementId which binds the messageId? Sorry about the ignorance, just need to understand what the link is :)

Reply:
The crux of this is that you're looking at the routing layer.

In v2, we've not really made a distinction between a departure and what will be an arrival id (both will be `MovementId`), because they're both short UUIDs and I'd imagine that some of the other services may be completely agnostic to them - indeed, it's entirely possible that transit-movements won't care about the distinction, so our connector won't either. 

As a result, `departureId` is just the name of the parameter that we're using in the URL, it will bind to a `MovementId` - this is mostly because we're trying to keep the URLs the same as v1 where possible.

We have this slightly complicated mess of binding because of a significant difference between v1 and v2 (and which is why I'm simply binding to a string here and thus in the routes), v1 uses integers as the departure/message IDs, v2 uses hex strings for the movement/message IDs. Because we want to use the same endpoint, we can't bind the parameters to an int or a hex string model, so instead I effectively say "I'm going to take over the binding phase, give me a string and I'll do the appropriate conversions".

Reply:
I've just been looking around the code and what we're doing and it looks like transit-movements will make a distinction, right down to the fact the URL has "departure" in there.

So that's a good question we need to ask - is `MovementId` something we should keep, or should be split it to `DepartureId` and later on `ArrivalId`?

Reply:
I was chatting with Rob, we've made it a `DepartureId` to be consistent.